wpengine/faustjs
# Fix regression in exit codes in Faust CLI

## Tasks


## Description

There was a regression in throwing the proper exit codes upon an empty WordPress URL and the GraphQL endpoint not being accessible. This PR fixes the errors by throwing the proper `1` exit code.

Technically, in the two cases (Where a `NEXT_PUBLIC_WORDPRESS_URL` env var doesn't exist or when the GraphQL Endpoint is unresponsive), we are handling the errors and providing the user with a message as to what went wrong. In some contexts, a `0` exit code then could be considered correct. However, since we are using this as a build tool, we want platforms like Atlas, Vercel, etc to know when these things go wrong, so throwing a `1` exit code makes more sense, since these are "blocking" errors and should technically fail a build/deployment.



## Related Issue(s):



## Testing

1. In the `canary` branch, build the packages `npm run build`
2. Delete the getting started example project's `.env.local` file
3. Run `npm run dev -w @faustwp/getting-started-example` and notice the `Could not find NEXT_PUBLIC_WORDPRESS_URL environment variable.` error message.
4. Write `echo $?` in the terminal to get the last command's exit code and notice it is `0`.
5. Repeat the steps in the `use-proper-exit-codes` branch and notice the proper `1` exit code.

[More info on exit codes](https://askubuntu.com/a/892605)



## Screenshots

Before:
<img width="1036" alt="Screenshot 2023-09-12 at 4 05 01 PM" src="https://github.com/wpengine/faustjs/assets/5946219/ad902a08-327a-404e-b4ad-80fe6daa8117">

After:
<img width="1101" alt="Screenshot 2023-09-12 at 4 04 40 PM" src="https://github.com/wpengine/faustjs/assets/5946219/7e3b1276-bfa1-4e83-8247-f148bb5d8950">





## Documentation Changes



## Dependant PRs
